docs/core_concepts/11_persistent_storage/index.md

```diff
@@ -0,0 +1,283 @@
+# Persistent Storage
+
+Persistent storage refers to any method of storing data that remains intact and accessible even after a system is powered off, restarted, or experiences a crash.
+
+In the context of Windmill, the stakes are: **where to store the data manipulated by Windmill** (ETL, data ingestion and preprocessing, data migration and sync etc.)?
+
+:::info TLDR
+
+Within Windmill, it is recommended to only store Windmill-specific elements ([resources](../3_resources_and_types/index.md), [variables](../2_variables_and_secrets/index.md) etc.). For data, we suggest using storage service providers to be targeted from Windmill.
+
+<br/>
+
```

Comment:
Add a paragraph along the lines.

There are 4 kinds of persistent storage:
- small data that is relevant in between script/flow execution and can be persistent on windmill itself
- big structured SQL data that is critical to your services and that is stored externally on an SQL Database or Warehouse
- object storage for large data such as S3
- NoSQL and document database such as MongoDB and Key-Value stores.

We will see an overview of all 4 below.